File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int a = Arrays.stream(nums).max().getAsInt();
        long res = 0;
        int start = 0, curr = 0;
        for (int num : nums) {
            curr += num == a ? 1 : 0;
            while (curr >= k) {
                curr -= nums[start++] == a ? 1 : 0;
            }
            res += start;
        }
        return res;
    }
}

File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.kt
class Solution {
    fun countSubarrays(nums: IntArray, k: Int): Long {
        var res = 0L
        val mostAppeared = Arrays.stream(nums).max().asInt
        var start = 0
        var curr = 0
        for (element in nums) {
            curr += if (element == mostAppeared) 1 else 0
            while (curr >= k) 
                curr -= if (nums[start++] == mostAppeared) 1 else 0
            res += start
        }
        return res
    }
}